pkg/appregistry/appregistry.go

```diff
@@ -14,21 +15,43 @@ func NewClientFactory() ClientFactory {
 	return &factory{}
 }
 
+type Options struct {
+	// Source refers to the URL of the remote app registry server.
+	Source string
+
+	// AuthToken refers to the authorization token required to access operator
+	// manifest in private repositories.
+	//
+	// If not set, it is assumed that the remote registry is public.
+	AuthToken string
+}
+
+// ClientFactory is an interface that wraps the New method.
+//
+// New returns a new instance of appregistry Client from the specified source.
 type ClientFactory interface {
-	// New returns a new instance of appregistry Client from given source and type
-	New(sourceType, source string) (Client, error)
+	New(options Options) (Client, error)
 }
 
 type factory struct{}
 
-func (f *factory) New(sourceType, source string) (Client, error) {
-	u, err := url.Parse(source)
+func (f *factory) New(options Options) (Client, error) {
+	u, err := url.Parse(options.Source)
 	if err != nil {
 		return nil, err
 	}
 
 	transport := httptransport.New(u.Host, u.Path, []string{u.Scheme})
 	transport.Consumers["application/x-gzip"] = runtime.ByteStreamConsumer()
+
+	// If a bearer token has been specified then we should pass it along in the headers
+	if options.AuthToken != "" {
+		fmt.Println("HAVE AUTH TOKEN")
+		fmt.Println(options.AuthToken)
```

Comment:
I am guessing you forgot to remove these before pushing. Please remove.

Reply:
whoops. fixed.